auction-app-frontend/src/pages/Shop/Shop.jsx

```diff
@@ -1,5 +1,159 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate, useLocation } from "react-router-dom";
+
+import { 
+  Button, 
+  Checkbox, 
+  ProductGrid,
+  ErrorComponent,
+  LoadingComponent 
+} from "src/components";
+
+import {
+  getProductsPaginated,
+  getCategoriesWithSubcategories
+} from "src/services";
+import { collapse, expand } from "src/assets/icons";
+
+import "./style.scss";
+
+const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
+
 const Shop = () => {
-  return <div>Shop Page</div>;
+  const [activeCategory, setActiveCategory] = useState();
+  const [items, setItems] = useState([]);
+  const [page, setPage] = useState(0);
+  const [categories, setCategories] = useState([]);
+  const [hasMore, setHasMore] = useState(true);
+  const [productsError, setProductsError] = useState(null);
+  const [productsLoading, setProductsLoading] = useState(true);
+  const [categoriesError, setCategoriesError] = useState(null);
+  const [categoriesLoading, setCategoriesLoading] = useState(true);
+  const [checked, setChecked] = useState({});
+
+  const navigate = useNavigate();
+  const query = useQuery();
+
+  const categoryId = query.get("category");
+  const searchProduct = query.get("search_product");
+
+  const fetchProducts = () => {
+    setProductsLoading(true);
+    
+    getProductsPaginated(page, 9, categoryId, searchProduct)
```

Comment:
This `9` we can move to constants and name it let's say DEFAULT_PAGE_NUMBER, so in case we want later to change number of products per page, we can do it in one place, and also avoid hard-coding values here.